### PR TITLE
[glib]: bump version to 2.74.0

### PR DESCRIPTION
**Pull request by theartful**
**Describe the pull request**

Update glib to 2.74.0. Glib now uses git submodules for gvdb, and `vcpkg_from_gitlab` does not fetch submodules, so I resorted to `vcpkg_download_distfile`.
I don't know of any other library that uses gvdb, so I think it's okay not to create an entire recipe for it.

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@hesham-essam ,Thanks for your pr, when I tested the feature by command "./vcpkg install glib[*]:x64-linux",
I got this error, could you please take a look?
````
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: running: autopoint --force
autoreconf: running: aclocal --force -I m4
autoreconf: configure.ac: tracing
autoreconf: running: libtoolize --copy --force
autoreconf: configure.ac: not using Intltool
autoreconf: running: gtkdocize --copy
Can't exec "gtkdocize": No such file or directory at /usr/share/autoconf/Autom4te/FileUtils.pm line 293.
autoreconf: error: gtkdocize failed with exit status: 2

````

**Comment by dg0yt** [comment]
> ~~~
> autoreconf: running: gtkdocize --copy
> Can't exec "gtkdocize": No such file or directory at /usr/share/autoconf/Autom4te/FileUtils.pm line 293.

Since vcpkg doesn't care for docs, it doesn't need `gtkdocize`. You may turn this step into a no-op by adding this to the portfile:
~~~
set(ENV{GTKDOCIZE} true)
~~~
Cf. libidn2, libtasn1.

**Comment by JonLiu1993** [comment]
I add `set(ENV{GTKDOCIZE} true)` to libmount portfile.cmake and fix the error, but got the another one:
Tested by command "./vcpkg install glib[*]:x64-linux",
````

Code:

        #ifdef __has_include
         #if !__has_include("afunix.h")
          #error "Header 'afunix.h' could not be found"
         #endif
        #else
         #include <afunix.h>
        #endif
Compiler stdout:

Compiler stderr:
 /home/test/test/vcpkg/buildtrees/glib/x64-linux-dbg/meson-private/tmp3bnbvpuy/testfile.c:4:12: error: #error "Header 'afunix.h' could not be found"
    4 |           #error "Header 'afunix.h' could not be found"
      |            ^~~~~

Has header "afunix.h" : NO

````

**Comment by dg0yt** [comment]
> Has header "afunix.h" : NO

On my linux machine, `afunix.h` comes with Ubuntu package `linux-headers`. (FTR this package is also needed by port `openssl`, and that port carries a reminder message.)

**Comment by hesham-essam** [comment]
@JonLiu1993
Glib does not havea  hard dependency on `afunix.h`. I get the same message as well, but glib builds without problems all the same (and vcpkg doesn't complain either):
```
Code:
 
        #ifdef __has_include
         #if !__has_include("afunix.h")
          #error "Header 'afunix.h' could not be found"
         #endif
        #else
         #include <afunix.h>
        #endif
Compiler stdout:
 
Compiler stderr:
 /home/theartful/Programming/vcpkg/buildtrees/glib/x64-linux-dbg/meson-private/tmpfr_q1e90/testfile.c:4:12: error: #error "Header 'afunix.h' could not be found"
    4 |           #error "Header 'afunix.h' could not be found"
      |            ^~~~~

Has header "afunix.h" : NO 
```

**Comment by hesham-essam** [comment]
Are you sure this is the cause of the build failure?

**Comment by JonLiu1993** [comment]
> Are you sure this is the cause of the build failure?

@hesham-essam , I use command "./vcpkg install glib[*]:x64-linux" tested again, 
this is error log, please take a look:

````
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: running: autopoint --force
autoreconf: running: aclocal --force -I m4
autoreconf: configure.ac: tracing
autoreconf: running: libtoolize --copy --force
autoreconf: configure.ac: not using Intltool
autoreconf: running: gtkdocize --copy
Can't exec "gtkdocize": No such file or directory at /usr/share/autoconf/Autom4te/FileUtils.pm line 293.
autoreconf: error: gtkdocize failed with exit status: 2

````

**Comment by hesham-essam** [comment]
@JonLiu1993  I removed my gtkdocize, reproduced the issue, then applied @dg0yt suggested solution, and things should work now. Please try again.

File: versions/l-/libmount.json [hunk 1]
```diff
@@ -1,7 +1,12 @@
 {
   "versions": [
     {
-      "git-tree": "bf368ca79f716bd0b287bf1eeedd87c3529831cf",
+      "git-tree": "a006357933a743ef48b94ce80a0565a1ce608769",
+      "version": "2.38",
+      "port-version": 1
+    },
+    {
+      "git-tree": "2bbde50cef7183b080b71e42f3adc7e6940adf12",
```
Comment (JonLiu1993) [inline versions/l-/libmount.json:9]:
```suggestion
      "git-tree": "bf368ca79f716bd0b287bf1eeedd87c3529831cf",
```

Comment (JonLiu1993) [reply versions/l-/libmount.json:9]: Please don't change the history of `git-tree`

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by hesham-essam** [comment]
@JonLiu1993  There are not really errors. Meson performs tests to check if certain functions exist. I get the same stuff
```
Compiler stderr:
 /home/theartful/Programming/vcpkg/buildtrees/glib/x64-linux-dbg/meson-private/tmp68g4jd6k/testfile.c: In function 'main':
/home/theartful/Programming/vcpkg/buildtrees/glib/x64-linux-dbg/meson-private/tmp68g4jd6k/testfile.c:17:18: error: #error "__builtin_getvfsstat not found"
   17 |                 #error "__builtin_getvfsstat not found"
      |                  ^~~~~

Checking for function "getvfsstat" : NO 
```
The cause of the error is something else. Can you please send me the log files?

**Comment by JonLiu1993** [comment]
> @JonLiu1993 These are not really errors. Meson performs tests to check if certain functions exist. I get the same stuff
> 
> ```
> Compiler stderr:
>  /home/theartful/Programming/vcpkg/buildtrees/glib/x64-linux-dbg/meson-private/tmp68g4jd6k/testfile.c: In function 'main':
> /home/theartful/Programming/vcpkg/buildtrees/glib/x64-linux-dbg/meson-private/tmp68g4jd6k/testfile.c:17:18: error: #error "__builtin_getvfsstat not found"
>    17 |                 #error "__builtin_getvfsstat not found"
>       |                  ^~~~~
> 
> Checking for function "getvfsstat" : NO 
> ```
> 
> The cause of the error is something else. Can you please send me the log files?


The error log file Error log I looked into and found a useful piece of information:
````
../src/glib-2-22354247da/meson.build:2123:2: ERROR: Dependency "libselinux" not found, tried pkgconfig and cmake

````

**Comment by hesham-essam** [comment]
@JonLiu1993 Well, to enable selinux support, you have to have libselinux installed. If you're on ubuntu, try `sudo apt install libselinux1-dev`

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**